pkg/cluster/delete.go

```diff
@@ -441,6 +441,12 @@ func (m *manager) Delete(ctx context.Context) error {
 		return err
 	}
 
+	m.log.Printf("deleting OIDC configuration")
+	err = m.rpBlob.DeleteBlobContainer(ctx, m.env.ResourceGroup(), m.env.OIDCStorageAccountName(), env.OIDCBlobContainerPrefix+m.doc.ID)
```

Comment:
For service principal clusters, wouldn't this return an error?

Reply:
Thanks, yes this needs the same condition as during creation. Will change it after the remaining testing it in CI e2e.

Reply:
Added the condition for the MIWI cluster for OIDC deletion.